src/common/bpf/distribution.rs

```diff
@@ -22,51 +23,62 @@ use super::*;
 pub struct Distribution<'a> {
     _map: &'a libbpf_rs::Map,
     mmap: memmap2::MmapMut,
-    prev: [u64; HISTOGRAM_BUCKETS],
-    heatmap: &'static Heatmap,
+    buffer: [u64; HISTOGRAM_BUCKETS],
```

Comment:
Shall we move this allocation to inside the else? Obviously it would suck it we are hitting that codepath, but since we expect never to, would end up saving some memory.

Reply:
This line doesn't allocate? The allocation happens at L#44

Reply:
Sorry, that was imprecise. I was suggesting moving this out of the structure altogether and onto L#60 since, afaict, the buffer is only used in the misaligned case.